mav_trajectory_generation/include/mav_trajectory_generation/impl/polynomial_optimization_nonlinear_impl.h

```diff
@@ -471,6 +490,58 @@ PolynomialOptimizationNonLinear<_N>::evaluateMaximumMagnitudeAsSoftConstraint(
   return cost;
 }
 
+template <int _N>
+void
+PolynomialOptimizationNonLinear<_N>::setFreeEndpointDerivativeHardConstraints(
+        const Vertex::Vector& vertices,
+        const std::vector<double>& initial_solution,
+        std::vector<double>* lower_bounds, std::vector<double>* upper_bounds) {
+  CHECK_NOTNULL(lower_bounds);
+  CHECK_NOTNULL(upper_bounds);
+  CHECK(lower_bounds->empty()) << "Lower bounds not empty!";
+  CHECK(upper_bounds->empty()) << "Upper bounds not empty!";
+
+  const size_t n_free_constraints = poly_opt_.getNumberFreeConstraints();
+  const size_t dim = poly_opt_.getDimension();
+  const int derivative_to_optimize = poly_opt_.getDerivativeToOptimize();
+
+  LOG(INFO) << "USE HARD CONSTRAINTS FOR ENDPOINT DERIVATIVE BOUNDARIES";
+
+  // Set all values to -inf/inf and reset only bounded opti param with values
+  for (const double x : initial_solution) {
+    lower_bounds->push_back(-HUGE_VAL);
```

Comment:
Just do this with a vector resize instead, and use std::numeric_limits instead of HUGE_VAL (which varies in implementation from compiler to compiler).
http://en.cppreference.com/w/cpp/container/vector/resize